vsintegration/tests/FSharp.Editor.Tests/CodeFixes/CodeFixTestFramework.fs

```diff
@@ -17,118 +17,273 @@ open FSharp.Editor.Tests.Helpers
 
 type TestCodeFix = { Message: string; FixedCode: string }
 
+module TestCodeFix =
+    /// Creates a test code fix from the given Roslyn source text and F# code fix.
+    let ofFSharpCodeFix (sourceText: SourceText) (codeFix: FSharpCodeFix) =
+        {
+            Message = codeFix.Message
+            FixedCode = string (sourceText.WithChanges codeFix.Changes)
+        }
+
 type Mode =
     | Auto
     | WithOption of CustomProjectOption: string
     | WithSignature of FsiCode: string
     | Manual of Squiggly: string * Diagnostic: string
     | WithSettings of CodeFixesOptions
 
-let inline toOption o =
-    match o with
-    | ValueSome v -> Some v
-    | _ -> None
-
-let mockAction =
-    Action<CodeActions.CodeAction, ImmutableArray<Diagnostic>>(fun _ _ -> ())
-
-let parseDiagnostic diagnostic =
-    let regex = Regex "([A-Z]+)(\d+)"
-    let matchGroups = regex.Match(diagnostic).Groups
-    let prefix = matchGroups[1].Value
-    let number = int matchGroups[2].Value
-    number, prefix
-
-let getDocument code mode =
-    match mode with
-    | Auto -> RoslynTestHelpers.GetFsDocument code
-    | WithOption option -> RoslynTestHelpers.GetFsDocument(code, option)
-    | WithSignature fsiCode -> RoslynTestHelpers.GetFsiAndFsDocuments fsiCode code |> Seq.last
-    | Manual _ -> RoslynTestHelpers.GetFsDocument code
-    | WithSettings settings -> RoslynTestHelpers.GetFsDocument(code, customEditorOptions = settings)
-
-let getRelevantDiagnostics (document: Document) =
-    cancellableTask {
-        let! _, checkFileResults = document.GetFSharpParseAndCheckResultsAsync "test"
+module ValueOption =
+    let inline toOption o =
```

Comment:
I did try that, but it didn't work because they're marked `inline` (even though FSharp.Editor has its internals visible to FSharp.Editor.Tests):

```
1>D:\src\fsharp\vsintegration\src\FSharp.Editor\Common\Extensions.fs(325,16): error FS1113: The value 'ofOption' was marked inline but its implementation makes use of an internal or private function which is not sufficiently accessible
1>D:\src\fsharp\vsintegration\tests\FSharp.Editor.Tests\CodeFixes\CodeFixTestFramework.fs(89,12): warning FS1116: A value marked as 'inline' has an unexpected value
1>D:\src\fsharp\vsintegration\tests\FSharp.Editor.Tests\CodeFixes\CodeFixTestFramework.fs(89,12): error FS1118: Failed to inline the value 'ofOption' marked 'inline', perhaps because a recursive value was marked 'inline'
1>Done building project "FSharp.Editor.Tests.fsproj" -- FAILED.
```

It feels like `toOption` and `ofOption` should really just be on FSharp.Core's `ValueOption` module anyway :)